website/docs/dev_publishing.md

```diff
@@ -506,6 +506,67 @@ or the scene file was copy pasted from different context.
 #### *Known errors*
 When there is a known error that can't be fixed by the user (e.g. can't connect to deadline service, etc.) `KnownPublishError` should be raised. The only difference is that its message is shown in UI to the artist otherwise a neutral message without context is shown.
 
+### Plugins
+Plugin is a single processing unit that can work with publish context and instances.
+
+#### Plugin types
+There are 2 types of plugins - `InstancePlugin` and `ContextPlugin`.  Be aware that inheritance of plugin from `InstancePlugin` or `ContextPlugin` actually does not affect if plugin is instance or context plugin, that is affected by argument name in `process` method.
```

Comment:
Is it still necessary to differentiate between those plugin parent classes if only what matter is name of input argument in process method? It seems to be confusing and redundant to support both at least.

Reply:
It's not something we can affect until we use `pyblish.base`. This is more likely warning for developers (as it is in dev docs)...

Reply:
> actually does not affect if plugin is instance or context plugin, that is affected by argument name in `process` method.

I'd actually argue that THIS is what is confusing in Pyblish, not the class names. It's an unfortunate side effect of Pyblish being backwards compatible that way.